src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -386,6 +384,18 @@ public SnowflakeStreamingIngestChannelInternal<?> openChannel(OpenChannelRequest
       // Add channel to the channel cache
       this.channelCache.addChannel(channel);
 
+      // Add encryption key to the client map for the table
+      if (response.getEncryptionKey() != null) {
+        this.encryptionKeysPerTable.put(
+            request.getFullyQualifiedTableName(),
```

Comment:
Names in Snowflake are quite complicated (see [ColumnNamesIT.java](https://github.com/snowflakedb/snowflake-ingest-java/blob/master/src/test/java/net/snowflake/ingest/streaming/internal/it/ColumnNamesIT.java#L20)), so I would avoid caching by names concatenated by `.`, it may lead to ambiguities. Instead, could we create a new class, which would represent the tuple `(databaseName, schemaName, tableName)` and implement `hashcode()` and `equals()` and use it as the caching key?